### PR TITLE
feat(github-app): use new dashboard <> engine api keys

### PR DESCRIPTION
**Pull request by sileht**
Fixes MRGFY-947

Change-Id: Iec3bbf896e3bc7d9b5516fa29662d7fe59d9b9a1

### CODE REVIEW

File: mergify_engine/clients/http.py [hunk 1]
```diff
@@ -206,7 +206,7 @@ def __init__(
         headers: typing.Optional[httpx_types.HeaderTypes] = None,
         timeout: httpx_types.TimeoutTypes = DEFAULT_TIMEOUT,
         base_url: httpx_types.URLTypes = "",
-    ):
+    ) -> None:
```
Comment (jd) [inline mergify_engine/clients/http.py:209]: could be in a different pr but ok :)

File: mergify_engine/config.py [hunk 2]
```diff
@@ -164,7 +174,7 @@ def ApplicationAPIKeys(v: str) -> typing.Dict[str, ApplicationAPIKey]:
         voluptuous.Required(
             "WEBHOOK_FORWARD_EVENT_TYPES", default=None
         ): voluptuous.Any(None, CommaSeparatedStringList),
-        # Mergify
+        # Mergify Engine tweaks
```
Comment (jd) [inline mergify_engine/config.py:177]: tweaks?

**Review by jd — APPROVED** [review]

File: mergify_engine/clients/http.py (hunk 1 above)
Comment (sileht) [reply mergify_engine/clients/http.py:209]: I don't get because mypy didn't see it before...

File: mergify_engine/config.py [hunk 3]
```diff
@@ -155,6 +159,12 @@ def ApplicationAPIKeys(v: str) -> typing.Dict[str, ApplicationAPIKey]:
         voluptuous.Required("APPLICATION_APIKEYS", default=""): voluptuous.Coerce(
             ApplicationAPIKeys
         ),
+        # Saas Special config
+        voluptuous.Required("ENGINE_TO_DASHBOARD_API_KEY"): str,
+        voluptuous.Required("DASHBOARD_TO_ENGINE_API_KEY"): str,
```
Comment (jd) [inline mergify_engine/config.py:164]: Will this block the startup of onprem?

Comment (sileht) [reply mergify_engine/config.py:164]: I have generated a random secret on startup for onpremise, so that will not block and make this APIs unusable.

**Review by jd — APPROVED** [review]
